src/Evolution/Systems/Cce/Components/KleinGordonCharacteristicEvolution.hpp

```diff
@@ -51,14 +51,65 @@ struct KleinGordonCharacteristicEvolution
   using cce_base = CharacteristicEvolution<Metavariables>;
   using typename cce_base::initialize_action_list;
 
+  template <typename BondiTag>
+  using hypersurface_computation =
+      typename cce_base::template hypersurface_computation<BondiTag>;
+
   using simple_tags_from_options =
       Parallel::get_simple_tags_from_options<initialize_action_list>;
 
   using typename cce_base::compute_scri_quantities_and_observe;
 
-  using self_start_extract_action_list = tmpl::list<>;
+  using self_start_extract_action_list = tmpl::list<
+      // note that the initialization will only actually happen on the
+      // iterations immediately following restarts
+      Actions::InitializeFirstHypersurface<
+          evolve_ccm, typename Metavariables::cce_boundary_component>,
+      tmpl::conditional_t<
+          tt::is_a_v<AnalyticWorldtubeBoundary,
+                     typename Metavariables::cce_boundary_component>,
+          Actions::UpdateGauge<false>, Actions::UpdateGauge<evolve_ccm>>,
+      Actions::PrecomputeGlobalCceDependencies,
+      tmpl::conditional_t<evolve_ccm,
```

Comment:
Do you plan on ever doing a KG CCM system?

Reply:
I don't have a plan for the next few months, but I can see two things that could motivate the KG CCM system. One is to simulate ringdown tails. The tail in the tensor sector (i.e., strain and $\Psi_4$) decays faster than the one in the scalar field. Therefore, digging out tails from the scalar field should be much easier than the tensor fields. If it turns out the SpEC CCM has a hard time doing the tensor tail, then I may have to switch to the scalar field first. 

The other one is related to Niko's worldtube projection. I have been talking with him about using this KG CCE system to extract GWs at future null infinity. But the usefulness of the CCM remains unclear.

So I feel it's good to keep those options in case they are needed.

Reply:
Ok that sounds like  good plan. Let's keep these compile-time switches in here, but they probably won't be used for a while.

You can squash your commits.